phoenix-core/src/main/java/org/apache/phoenix/compile/OffsetCompiler.java

```diff
@@ -62,23 +82,233 @@ public SortOrder getSortOrder() {
+            }
+
+            // Now that columns etc have been resolved lets check to make sure they match the pk order
+            // Today the RowValueConstuctor Equality gets Rewritten into AND of EQ Ops.
+            if (!(whereExpression instanceof AndExpression)) {
+                LOGGER.warn("Unexpected error while compiling RVC Offset, expected AndExpression got " + whereExpression.getClass().getName());
+                throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the tables PKs.");
+            }
+
+            List<RowKeyColumnExpression> rowKeyColumnExpressionList = buildListOfRowKeyColumnExpressions(whereExpression.getChildren(),isIndex);
+            if(rowKeyColumnExpressionList.size() != numUserColumns) {
+                LOGGER.warn("Unexpected error while compiling RVC Offset, expected " + numUserColumns + " found " + rowKeyColumnExpressionList.size());
+                throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the table's PKs.");
+            }
+
+            for (int i = 0; i < numUserColumns; i++) {
+                PColumn column = columns.get(i + userColumnIndex);
+                //
+                ColumnParseNode columnParseNode = rvcColumnParseNodeList.get(i);
+
+                String columnParseNodeString = columnParseNode.getFullName();
+                if (isIndex) {
+                    columnParseNodeString = IndexUtil.getDataColumnName(columnParseNodeString);
+                }
+
+                RowKeyColumnExpression rowKeyColumnExpression = rowKeyColumnExpressionList.get(i);
+                String expressionName = rowKeyColumnExpression.getName();
+
+                // Not sure why it is getting quoted
+                expressionName = expressionName.replace("\"", "");
+
+                if (isIndex) {
+                    expressionName = IndexUtil.getDataColumnName(expressionName);
+                }
+
+                if (!StringUtils.equals(expressionName, columnParseNodeString)) {
+                    throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the tables PKs.");
+                }
+
+                //
+                String columnString = column.getName().getString();
+                if (isIndex) {
+                    columnString = IndexUtil.getDataColumnName(columnString);
+                }
+                if (!StringUtils.equals(expressionName, columnString)) {
+                    throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the tables PKs.");
+                }
+            }
+
+            // check to see if this was a single key expression
+            ScanRanges scanRanges = context.getScanRanges();
+
+            if (!scanRanges.isPointLookup()) {
+                throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must be a point lookup.");
```

Comment:
I'm not sure we should bubble that up to the user as is. If it doesn't resolve to a point lookup, that means something went wrong with the Phoenix client-side logic right? Since, we enforce:
1. Must use all of the PK columns
2. Must be convertible to a conjunction of PK=x equalities

So, this already means it will be a point lookup. Bubbling this up to the user won't be helpful IMO since it sort of might throw them off indicating that they are doing something wrong, but I'm open to discussion.

Reply:
I think the most likely reason this goes wrong is that the user provides something strange in the offset.  I agree its more of a sanity check.  But i'm open to either.